### ISSUE TITLE
Return whether new value will be processed

### ISSUE THREAD

**Issue by bartbutenaers** [bnielsen1965/button-events#1]
Morning Bryan,

I have finalized my integration of your library for Node-RED, and included some timelines on my [readme](https://github.com/bartbutenaers/node-red-contrib-button-events) page.  Just mentioning it here, in case somebody else is struggling to understand and finetune the 3 time intervals...

We are now [discussing ](https://discourse.nodered.org/t/announce-node-red-contrib-button-events-beta-preview) my beta version on the Node-RED forum, which resulted in a feature request...
BTW The reactions are very positive, so your library will be a VERY nice additions to our IOT framework!!!!

When I pass a new value to your `gpioChange`, I don't know whether it will processed or whether it is rejected (due to the debounce interval or disabled state).  That could be solved e.g. by returning some kind of feedback, for example:
```
(value) {
    if (this.buttonState === STATE_DISABLED) return "disabled";
    value = (this.Config.usePullUp ? !value : value); // invert if pull up
    this.lastValue = value;
    if (!this.debounce) {
          this.debounceStart(value);
          return "accepted";
    }
    
   return "debounced";
}
```
I have not created a pull-request for this, since there are a number of options to solve it and I don't know your preference...

Thanks again!
Bart

**Comment by bnielsen1965** [comment]
I can add that. I was looking at the code and I think I have some leftover variables from an old algorith, this.currentValue doesn't appear to be used. I'm going to look at this a bit because I don't recall at the moment the original purpose of currentValue in the debounce.

Would you want a text response or would a boolean suffice as a logic switch to indicate that it is in a debounce state?

**Comment by bartbutenaers** [comment]
For me it doesn't matter if you want to return a string or an [enum](https://www.sohamkamani.com/blog/2017/08/21/enums-in-javascript/#enumerations-with-objects).  But I would prefer not to use a boolean, because then I can't show a reason why the value is ignored.  And perhaps you might add extra functionality in the future, so extra reasons might be introduced then ...

BTW In this [line](https://github.com/bnielsen1965/button-events/blob/master/index.js#L45) you have a typo in `deboucneTimer `.  It will work anyway, but it looks nicer if you fix it...

**Comment by bnielsen1965** [comment]
I reviewed the code in your previous comment again, I missed the *return "disabled"*, I now understand the reasoning for the string. I'm going to do exactly as you suggested.

I fixed the spelling bug in the constructor, as well as several spelling errors in the README, and removed that currentValue setting as it is not needed, it was leftover code from development and testing.

I noticed that the node red gpio also has a debounce setting so I made the button-events disable the debounce timer if the confguration has a timing.debounce === 0. This also means there is an additional gpioChange() status value, I used "final" if debounce is disabled. So the possible gpioChange() status return strings are...

- **"disabled"**, The button event handler is disabled and will not process gpioChange() values.
- **"debounced"**, The debounce timer is running and the input value has been updated with the last gpioChange() value.
- **"accepted"**, The gpioChange() value has been accepted as the starting value and the debounce timer started.
- **"final"**, The debounce timer is disabled and the gpioChange() value is accepted as the final value.


I have not published yet, I'm going to do some testing first.


EDIT: I added a little test.js script to the project, everything looks good with and without debounce. Let me know if you see anything missing, once it looks good to you  I'll hit publish.

**Comment by bartbutenaers** [comment]
Hi Bryan,
I have published my node on [npm](https://www.npmjs.com/package/node-red-contrib-button-events).  I have also tested your status returns strings, and it works very well!  It offered me the opportunity to add an extra feature to my node.
So for me this is all working well!  
As soon as you publish your new version on npm, my users will have the new feature available ...
Have a nice weekend!
Bart
